packages/client/src/features/smart-connector/smart-connector.ts

```diff
@@ -0,0 +1,524 @@
+            }
+            else if (!this.groupIsCollapsed[parent.id] && header.previous()) {(groupItems.last()).focus();}
+        }
+        if (matchesKeystroke(event, this.nextElementKeyCode)) {
+            if ((this.groupIsCollapsed[parent.id] || this.groupIsTop[parent.id]) && parent.next()) {
+                const collapsableHeader = getHeaderIfGroupContainsCollapsable(parent.next());
+                if (collapsableHeader &&
+                    (!this.groupIsTop[parent.next().id] || this.groupIsCollapsed[parent.next().id])) {collapsableHeader.focus();}
+                else {this.getNextGroupFirstItem(parent).focus();}
+            }
+            else if (!this.groupIsCollapsed[parent.id] && header.next()) {(groupItems.first()).focus();}
+        }
+    }
+
+    protected toolButtonNavigation(event: KeyboardEvent, item: PaletteItem): void {
+        const parentId = this.smartConnectorItems.find(e => e.children?.includes(item))!.id;
+        const parent = document.getElementById(parentId)!;
+        const toolButton = document.getElementById(item.id)!;
+        const collapsableHeader = getHeaderIfGroupContainsCollapsable(parent);
+        if (matchesKeystroke(event, this.previousElementKeyCode)) {
+            const previousGroupCollapsableHeader = getHeaderIfGroupContainsCollapsable(parent.previous());
+            if (toolButton.previous()) {toolButton.previous().focus();}
+            else if (collapsableHeader && !this.groupIsTop[parent.id]) {collapsableHeader.focus();}
+            else if (previousGroupCollapsableHeader && this.groupIsTop[parent.previous().id]) {previousGroupCollapsableHeader.focus();}
+            else if (parent.previous()) {this.getPreviousGroupLastItem(parent).focus();}
+        }
+        if (matchesKeystroke(event, this.nextElementKeyCode)) {
+            const nextGroupCollapsableHeader = getHeaderIfGroupContainsCollapsable(parent.next());
+            if (toolButton.next()) {toolButton.next().focus();}
+            else if (collapsableHeader && this.groupIsTop[parent.id]) {collapsableHeader.focus();}
+            else if (nextGroupCollapsableHeader && !this.groupIsTop[parent.next().id]) {nextGroupCollapsableHeader.focus();}
+            else if (parent.next()) {this.getNextGroupFirstItem(parent).focus();}
+        }
+    }
+
+    protected getNextGroupFirstItem(parent: HTMLElement): HTMLElement {
+        return parent.nextElementSibling!.getElementsByClassName(GROUP_CLASS)[0].firstElementChild as HTMLElement;
+    }
+
+    protected getPreviousGroupLastItem(parent: HTMLElement): HTMLElement {
+        return parent.previousElementSibling!.getElementsByClassName(GROUP_CLASS)[0].lastElementChild as HTMLElement;
+    }
+
+    // #endregion
+
+    protected onClickCreateToolButton(_button: HTMLElement, item: PaletteItem)  {
+        return (_ev: MouseEvent) => {
+            this.createTool(item);
+        };
+    }
+
+    protected createTool(item: PaletteItem): void {
```

Comment:
```suggestion
    protected triggerCreation(item: PaletteItem): void {
```

Or something similar?